### ISSUE TITLE
ODRL Ontology in non-perfect state

### ISSUE THREAD

**Issue by vroddon** [w3c/poe#189]
The current ODRL ontology fails to be loaded by Protègè v.5.2 (the up to date version).

I cannot openly denounce any specific mistake, but I raise the warning as most of our readers will use Protègè. Or am I the only one with such a problem? I have been able to open the ontology with specific software (owlgred, swoop and older versions of Protègè once transformed the ontology into RDF/OWL by converters like EasyRDF -http://www.easyrdf.org/converter-)

My own wrapper around OWL API (available here http://owlprofiler.appspot.com/) actually determines the ontology is *not OWL DL* because it is using several undefined resources. 
For example, we speak about vcard:Organization, but the reasoner does not have any clue about it being a class, and consistently throws an error. The same applies to the metadata properties we use ("Use of undeclared object property"). We can solve them, I think. For example, once I simply add the axiom:

`vcard:Organization a owl:Class .
`
The error promptly disappears. I believe the same policy should be followed by each of the external elements. 

The inconsistencies are also recognized by Pellet, but not by the online validator of the Univ. of Manchester. http://mowl-power.cs.man.ac.uk:8080/validator/
The ultimate goal I have in mind is that nobody can claim the ontology has errors/inconsistencies. Any opinion about this?

**Comment by iherman** [comment]
> The current ODRL ontology fails to be loaded by Protègè v.5.2 (the up to date version).
> 
> I cannot openly denounce any specific mistake, but I raise the warning as most of our readers will use Protègè. Or am I the only one with such a problem? I have been able to open the ontology with specific software (owlgred, swoop and older versions of Protègè once transformed the ontology into RDF/OWL by converters like EasyRDF -http://www.easyrdf.org/converter- <http://www.easyrdf.org/converter->)
> 
> My own wrapper around OWL API (available here http://owlprofiler.appspot.com/ <http://owlprofiler.appspot.com/>) actually determines the ontology is not OWL DL because it is using several undefined resources.
> For example, we speak about vcard:Organization, but the reasoner does not have any clue about it being a class, and consistently throws an error. The same applies to the metadata properties we use ("Use of undeclared object property"). We can solve them, I think. For example, once I simply add the axiom:
> 
> vcard:Organization a owl:Class .
> The error promptly disappears. I believe the same policy should be followed by each of the external elements.
> 
> The inconsistencies are also recognized by Pellet, but not by the online validator of the Univ. of Manchester. http://mowl-power.cs.man.ac.uk:8080/validator/ <http://mowl-power.cs.man.ac.uk:8080/validator/>
> The ultimate goal I have in mind is that nobody can claim the ontology has errors/inconsistencies. Any opinion about this?
> 

While I think having our ontology recognized by these tools is a fine goal, I am a bit uneasy. To take the vcard example above: as a naïve user I would expect a reasoning tool to pick up the ontology for vcard and add the resulting axioms to the ODRL ones, and not expect everything to be in one place. Otherwise the tools breaks the distributive nature of the Web.

Of course, if the vcard (OWL) ontology does not exist, or is not accurate than we may think of doing this but then… again thinking in an idealistic semantic web way, we add a new statement on an ontology that we do not own and that statement will be available for everyone on the (Semantic) Web. It is not really o.k., is it? (I believe community referred to that as 'ontology hijacking':-)

Bottom line: I am tempted not to add these external elements.

**Comment by vroddon** [comment]
Well, in the particular case of vcard, the ontology defines the Organization as an owl:Class (https://www.w3.org/2006/vcard/ns#Organization) but for some reason Protègè fails to obtain it --my guess is that the ontology URI  starts with http: whereas the redirection points to https:.

But of course, the case you explain may be the case of other vocabularies we use. Before raising the issue publicly I will study the case more in detail.

**Comment by iherman** [comment]
> On 5 Jun 2017, at 13:46, Víctor Rodríguez-Doncel <notifications@github.com <mailto:notifications@github.com>> wrote:
> 
> Well, in the particular case of vcard, the ontology defines the Organization as an owl:Class (https://www.w3.org/2006/vcard/ns#Organization <https://www.w3.org/2006/vcard/ns#Organization>) but for some reason Protègè fails to obtain it --my guess is that the ontology URI  starts with http: whereas the redirection points to https:.
> 

Ah. It may indeed be a matter of http vs https. I would think that is a problem that Protege has to solve, and it is not an ODRL problem. Ie, I do not think we have to deal with it.

> But of course, the case you explain may be the case of other vocabularies we use. Before raising the issue publicly I will study the case more in detail.
> 

Thanks

**Comment by riannella** [comment]
I  loaded https://raw.githubusercontent.com/w3c/poe/gh-pages/vocab/ODRL22.ttl
into Protege 5.1 and 5.2 and could not see any errors?

**Comment by vroddon** [comment]
I downloaded the software again and this time I had no trouble. 
Newer reasoners do not complain, probably they follow the redirects correctly.
I also tried the Ontology pitfall scanner "Oops" (http://oops.linkeddata.es/). It only reveled what we know ("http://www.w3.org/ns/odrl/2/hasPolicy may not be inverse of http://www.w3.org/ns/odrl/2/target ") plus a few minor issues.
